src/gen_smtp_server_session.erl

```diff
@@ -1405,6 +1416,32 @@ report_recipient(multiple, [{ResponseType, Value} | Rest], State) ->
     report_recipient(ResponseType, Value, State),
     report_recipient(multiple, Rest, State).
 
+% parse proxy protocol
+% http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+
+read_proxy_protocol(Transport, Socket) ->
+    ok = Transport:setopts(Socket, [{packet, line}, binary]),
+    case Transport:recv(Socket, 0, 5000) of
+        {ok, Data} ->
+            case Data of
+                <<"PROXY ", _/binary>> ->
+                    case binary:split(Data, <<" ">>, [global]) of
+                        [<<"PROXY">>, InetFamily, SrcIp, _DstIp, _SrcPort, _DstPort] when InetFamily == <<"TCP4">> orelse InetFamily == <<"TCP6">> ->
```

Comment:
You should probably use the right, address specific parser for v4 or v6, so if the proxy sends TCP4 with a v6 address, things fail.

Reply:
Huh... can you be more explicit ?